flax/core/scope.py

```diff
@@ -329,9 +329,10 @@ def reserve(self, name: str):
       name: the name to reserve.
     """
     if not isinstance(name, str):
-      raise errors.ScopeNameTypeError(name)
+      raise TypeError('The type of scope "{name}" should be string but '
+                     f'it is {type(name)}')
     if name in self.reservations:
-      raise errors.ScopeNameInUseError(name)
+      raise ValueError(f'Duplicate use of scope name: "{name}"')
```

Comment:
Don't we lose a valuable docstring that existed on `ScopeNameInUseError`?

Reply:
Actually the docstring on `ScopeNameInUseError` describes the problem using Modules: https://flax.readthedocs.io/en/latest/flax.errors.html#flax.errors.ScopeNameInUseError

Therefore, I've now moved that description to `errors.NameInUseError` and made sure it is thrown in `module.py`. The error here is only thrown if a user is using Scopes directly, which isn't something the average user will do (I think).

Reply:
Ah I understand now. Makes sense.